src/components/Album/Album.js

```diff
@@ -1,39 +1,90 @@
-import './Album.css'
-import Form from '../Form/Form'
-import { getAlbumsByMasterId } from "../../api"
-import UserContext from "../UserContext/UserContext"
+import React, { useState, useEffect, useContext } from 'react';
+import { useParams } from 'react-router-dom';
+import { getAlbumDetails } from '../../api';
+import Form from '../Form/Form';
+import './Album.css';
+import UserContext from '../UserContext/UserContext';
 
 function Album() {
-  const {setCurrentUser} = useContext(UserContext)
-  const [album, setAlbum] = useState(null)
-  const [isLoading, setIsLoading] = useState(true)
-  const [modal, setModal] = useState(false)
-  
-  const {id} = useParams()
-  
-  const showModal = () => {
-    modal ? setModal(false) : setModal(true)
-  }
+  const { id } = useParams();
+  const [albumDetails, setAlbumDetails] = useState(null);
+  const [isLoading, setLoading] = useState(true);
+  const [modal, setModal] = useState(false);
+  const {currentUser, setCurrentUser} = useContext(UserContext) 
 
   useEffect(() => {
-    getAlbumsByMasterId(id)
-      .then(res => {
-        setAlbum(res)
-        setIsLoading(false)
+    setLoading(true);
+    getAlbumDetails(id)
+      .then(data => {
+        setAlbumDetails(data);
+        setLoading(false);
+      })
+      .catch(error => {
+        console.error('Error fetching album details:', error);
       })
-  }, [])
-
-  
-  if(!isLoading) {
-    return (
-      <>
-       <p>{id} hello</p>
-       {!modal && <button onClick={showModal}>Add a journal entry</button>}
-       {modal && <Form {...album} setCurrentUser={setCurrentUser} showModal={showModal}/>}
-      </>
-    )
-  } 
+      .finally(() => {
+        setLoading(false);
+      });
+  }, [id]);
+
+  const handleAddToCollections = () => {
+    const { id, title, artist, coverImg } = albumDetails
+
+    const newAlbum = {
+      masterId: id,
+      title: title,
+      artist: artist,
+      thumb: coverImg
+    }
+
+    setCurrentUser(prev => ({
+      ...prev,
+      collections: [...prev.collections, newAlbum]
+    }))
+  }
+
+  const showModal = () => {
+    setModal(prevModal => !prevModal);
+  };
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!albumDetails) {
+    return <div>Album not found.</div>;
+  }
+
+  const { title, artist, releaseDate, genre, styles, tracklist, coverImg } = albumDetails;
+
+  return (
+    <div>
+      <div className="buttons-container">
+        <button className="add-to-collections-button" onClick={() => handleAddToCollections()}>
+          Add to Collections
+        </button>
+        {!modal && <button className="journal-button" onClick={showModal}>Add to Journal Entry</button>}
+
+      </div>
+      <img classname="cover-image" src={coverImg} alt={`Cover art for ${title}`} />
+      <h2>{title}</h2>
+      <p>Artist: {artist}</p>
+      <p>Release Date: {releaseDate}</p>
+      <p>Genre: {genre}</p>
+      <p>Styles: {styles.join(', ')}</p>
```

Comment:
Here's where I was running into some Albums that didn't have this `styles` key
```suggestion
      {styles && <p>Styles: {styles.join(', ')}</p>} 
```

Reply:
{styles && styles.length > 0 && <p>Styles: {styles.join(', ')}</p>}

changed.